### PR TITLE
Use python3.8 for 'python on macos aarch64'

### PR DESCRIPTION
**Pull request by charliermarsh**

### CODE REVIEW

**Comment by zanieb** [comment]
Prioritizing because this fails in the cache check now https://github.com/astral-sh/uv/actions/runs/8491249792/job/23263080140

They must have changed the macOS 14 runners.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -271,7 +271,7 @@ jobs:
         run: curl -LsSf "https://github.com/astral-sh/uv/releases/latest/download/uv-aarch64-apple-darwin.tar.gz" | tar -xvz
 
       - name: "Check cache compatibility"
-        run: python scripts/check_cache_compat.py --uv-current ./uv --uv-previous ./uv-aarch64-apple-darwin/uv
+        run: python3.8 scripts/check_cache_compat.py --uv-current ./uv --uv-previous ./uv-aarch64-apple-darwin/uv
```
Comment (zanieb) [inline .github/workflows/ci.yml:274]: No real reason for this test  test to use brew's Python. We may want to change that.

**Review by zanieb — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (charliermarsh) [reply .github/workflows/ci.yml:274]: I think it’s intentional to have a test for Brew Pythons though.

Comment (zanieb) [reply .github/workflows/ci.yml:274]: Yeah but we don't need it in the cache compatibility test, just the system check.

**Comment by zanieb** [comment]
Reproduced this locally

```
❯ /opt/homebrew/bin/python3.9 scripts/check_system_python.py
INFO: Checking that `pylint` isn't installed.
WARNING: Package(s) not found: pylint
INFO: Installing the package `pylint`.
Resolved 7 packages in 5ms
Installed 1 package in 10ms
 + pylint==3.1.0
INFO: Checking that `pylint` is installed.
WARNING: Package(s) not found: pylint
Traceback (most recent call last):
  File "/Users/mz/workspace/uv/scripts/check_system_python.py", line 82, in <module>
    raise Exception("The package `pylint` isn't installed (but should be).")
Exception: The package `pylint` isn't installed (but should be).
```

**Comment by zanieb** [comment]
`/Library/Frameworks/Python.framework/Versions/3.12/bin/python3` is being used on my machine instead of the HomeBrew Python.

**Comment by zanieb** [comment]
Superseded by #2735 and #2736

**Pull request closed without merging**